src/AnnotationThread.js

```diff
@@ -496,14 +499,12 @@ class AnnotationThread extends EventEmitter {
      * @return {void}
      */
     updateTemporaryAnnotation(tempAnnotation, savedAnnotation) {
-        if (!(tempAnnotation.annotationID in this.annotations)) {
-            // If no temporary annotation is found, save to thread normally
-            this.saveAnnotationToThread(savedAnnotation);
-        } else {
-            // Otherwise, replace temporary annotation with annotation saved to server
-            delete this.annotations[tempAnnotation.annotationID];
-            this.annotations[savedAnnotation.annotationID] = savedAnnotation;
+        if (this.annotations.includes(tempAnnotation)) {
```

Comment:
Minor: Could you simplify this to `this.annotations = this.annotations.filter(annotation => annotation !== tempAnnotation)`?